starlette/requests.py

```diff
@@ -136,7 +136,7 @@ def cookies(self) -> typing.Dict[str, str]:
     @property
     def client(self) -> Address:
         host, port = self.scope.get("client") or (None, None)
-        return Address(host=host, port=port)
+        return Address(host=host, port=port)  # type: ignore[arg-type]
```

Comment:
Hmmm... This is actually not ASGI compliant: https://asgi.readthedocs.io/en/latest/specs/www.html#http-connection-scope

> client (Iterable[Unicode string, int]) – A two-item iterable of [host, port], where host is the remote host’s IPv4 or IPv6 address, and port is the remote port as an integer. Optional; if missing defaults to None.

Reply:
How is this not ASGI compliant? It's the same thing we had before. In fact, I don't think we even need the `# type: ignore`. I just removed it in 417a3d7.

Reply:
Because `Address` is not supposed to accept (`None`, `None`) it's actually `Optional[Address]`.

My bad anyway, I should have mentioned that it's not because of your PR... I just noticed that it's currently wrong.

Reply:
let's fix it in another PR then 😄

Reply:
Opened #1462